### PR TITLE
modified for spinning lidar

### PR DESCRIPTION
**Pull request by chengwei0427**
Hi, Dear friends,

I made some adaptations to support spingning lidar, and test with velodyne-16/ouster-64/ouster-32， the algo is amazing. Thank you for your contribution to the open source community.

### CODE REVIEW

File: config/params.yaml [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+  # GPS Settings
+  useImuHeadingInitialization: true           # if using GPS data, set to "true"
+  useGpsElevation: false                      # if GPS elevation is bad, set to "false"
+  gpsCovThreshold: 2.0                        # m^2, threshold for using GPS data
+  poseCovThreshold: 25.0                      # m^2, threshold for using GPS data
+  
+  # Export settings
+  savePCD: false                              # https://github.com/TixiaoShan/LIO-SAM/issues/3
+  savePCDDirectory: "/Downloads/LOAM/"        # in your home folder, starts and ends with "/". Warning: the code deletes "LOAM" folder then recreates it. See "mapOptimization" for implementation
```
Comment (wodlige) [inline config/params.yaml]: 没有在cpp文件中找到保存pcd文件相关的服务请求

Comment (chengwei0427) [reply config/params.yaml]: 抱歉，配置参数没有删除干净

File: src/lio/PoseEstimation.cpp [hunk 2]
```diff
@@ -514,107 +552,113 @@ void process(){
...
+  exTlb.topLeftCorner(3, 3) = R;
+  exTlb.topRightCorner(3, 1) = t;
   exRlb = R;
+
   exRbl = R.transpose();
   exPlb = t;
   exPbl = -1.0 * exRbl * exPlb;
 
-  ros::Subscriber subFullCloud = nodeHandler.subscribe<sensor_msgs::PointCloud2>("/livox_full_cloud", 10, fullCallBack);
+  ros::Subscriber subFullCloud = nodeHandler.subscribe<sensor_msgs::PointCloud2>("/full_cloud", 10, fullCallBack);
```
Comment (wodlige) [inline src/lio/PoseEstimation.cpp:641]: 在featureExtract节点并没有找到发布sensor_msgs::PointCloud2消息类型的/full_cloud话题，所以在PoseEstimation节点订阅的/full_cloud话题是哪个节点发布的呢？下面是featureExtract节点发布的四个话题：
pubCornerPoints = nh.advertise<sensor_msgs::PointCloud2>("/laser_cloud_edge", 1);
pubSurfacePoints = nh.advertise<sensor_msgs::PointCloud2>("/laser_cloud_surf", 1);
pubFullPoints = nh.advertise<sensor_msgs::PointCloud2>("/laser_cloud_filtered", 10);
pubLaserCloudInfo = nh.advertise<GC_LOAM::cloud_info>("/feature/cloud_info", 1);

Comment (chengwei0427) [reply src/lio/PoseEstimation.cpp:641]: 实际订阅的是“laser_cloud_filtered”这个topic的数据；

Comment (wodlige) [reply src/lio/PoseEstimation.cpp:641]: 所以得修改一下PoseEstimation节点订阅的话题名。

**Pull request closed without merging**